### PR TITLE
🌐 Update `Optional` usage in Chinese translation for Python Types Intro `docs/zh/docs/python-types.md`

### PR DESCRIPTION
**Pull request by maoyibo**
Add `Generic types with type parameters` and `Generic types`, Fix the code file name of `Optional` and subsequent chapters.

### CODE REVIEW

File: docs/zh/docs/python-types.md [hunk 1]
```diff
@@ -215,20 +221,43 @@ John Doe
     * 这个 `dict` 的所有键为 `str` 类型（可以看作是字典内每个元素的名称）。
     * 这个 `dict` 的所有值为 `float` 类型（可以看作是字典内每个元素的价格）。
 
+### `Optional`
+
+你也可以使用 `Optional` 来声明一个变量有一个类型，比如 `str`, 但是它是 "optional", 这意味着它也可以是 `None`:
```
Comment (waynerv) [inline docs/zh/docs/python-types.md:226]: 但是它是 "optional" --》但是它是「可选的」

File: docs/zh/docs/python-types.md [hunk 2]
```diff
@@ -215,20 +221,43 @@ John Doe
...
+#### 泛型类型
+
+这些类型采用方括号中的类型参数，如:
+
+* `List`
+* `Tuple`
+* `Set`
+* `Dict`
+* `Optional`
+* ...and others.
```
Comment (waynerv) [inline docs/zh/docs/python-types.md:243]: ...及其他。

**Review by waynerv — CHANGES_REQUESTED** [review]

**Review by waynerv — APPROVED** [review]

File: docs/zh/docs/python-types.md (hunk 1 above)
Comment (maoyibo) [reply docs/zh/docs/python-types.md]: Fixed, Thanks.

File: docs/zh/docs/python-types.md (hunk 2 above)
Comment (maoyibo) [reply docs/zh/docs/python-types.md]: Fixed, Thanks.

**Review by waynerv — APPROVED** [review]

**Comment by tiangolo** [comment]
Hum, as the branch was removed, I can't update this PR to see if there are any new changes, so I'll have to close it and wait for some new PR with the same (or other) changes. 😅

**Pull request closed without merging**